### PR TITLE
[NOT READY] Values and README Files Improvements

### PR DESCRIPTION
**Pull request by Xtigyro**
- Put warning for changing the default puppetserver port in `values.yaml`.
- Code style fixes in "values.yaml".
- Improve Testing section in `README.md`.

### CODE REVIEW

**Comment by Xtigyro** [comment]
CC: @slconley @scottcressi @mwaggett @nwolfe @adrienthebo @dhollinger @raphink @binford2k

**Comment by Xtigyro** [comment]
@underscorgan @Iristyle Please use the `Squash and merge` button upon merging.

File: values.yaml [hunk 1]
```diff
@@ -61,6 +61,9 @@ puppetserver:
       ## The LB type (network protocol) for some cloud providers must be set here.
       ports:
         puppetserver:
+          ## WARNING: if a different port is set, PuppetDB and Puppet Compiler/s container images
+          ## must be modified - so that they can connect to Puppet Master/s K8s Service and vice versa.
+          ## Their Docker entrypoint scripts expect port "8140".
```
Comment (underscorgan) [inline values.yaml:66]: I don't think this is accurate. In the puppetserver container there's the `PUPPET_MASTERPORT` environment variable and in the puppetdb container there's the `PUPPETSERVER_PORT` environment variable so you should be able to change those as long as the env variables are being set

Comment (Xtigyro) [reply values.yaml:66]: Thanks! I wasn't sure neither - therefore, wanted you to check it. 

Will fix it - so changing the ports becomes a working option.

**Comment by Xtigyro** [comment]
I'm going to open a fresh new PR which will include much more changes - and the chart version is going to be bumped to `v4.4.0`.

**Pull request closed without merging**